components/script/dom/stereopannernode.rs

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::dom::audioparam::AudioParam;
+use crate::dom::audioscheduledsourcenode::AudioScheduledSourceNode;
+use crate::dom::baseaudiocontext::BaseAudioContext;
+use crate::dom::bindings::codegen::Bindings::AudioNodeBinding::{
+    ChannelCountMode, ChannelInterpretation,
+};
+use crate::dom::bindings::codegen::Bindings::AudioParamBinding::AutomationRate;
+use crate::dom::bindings::codegen::Bindings::StereoPannerNodeBinding::StereoPannerNodeMethods;
+use crate::dom::bindings::codegen::Bindings::StereoPannerNodeBinding::{self, StereoPannerOptions};
+use crate::dom::bindings::error::{Error, Fallible};
+use crate::dom::bindings::reflector::reflect_dom_object;
+use crate::dom::bindings::root::{Dom, DomRoot};
+use crate::dom::window::Window;
+use dom_struct::dom_struct;
+use servo_media::audio::node::AudioNodeInit;
+use servo_media::audio::param::ParamType;
+use servo_media::audio::stereo_panner::StereoPannerOptions as ServoMediaStereoPannerOptions;
+use std::f32;
+
+#[dom_struct]
+pub struct StereoPannerNode {
+    source_node: AudioScheduledSourceNode,
+    pan: Dom<AudioParam>,
+}
+
+impl StereoPannerNode {
+    #[allow(unrooted_must_root)]
+    pub fn new_inherited(
+        window: &Window,
+        context: &BaseAudioContext,
+        options: &StereoPannerOptions,
+    ) -> Fallible<StereoPannerNode> {
+        let node_options = options.parent.unwrap_or(
+            2,
+            ChannelCountMode::Clamped_max,
+            ChannelInterpretation::Speakers,
+        );
+        if node_options.mode == ChannelCountMode::Max {
+            return Err(Error::NotSupported);
+        }
+        if node_options.count > 2 || node_options.count == 0 {
+            return Err(Error::NotSupported);
+        }
+        let source_node = AudioScheduledSourceNode::new_inherited(
+            AudioNodeInit::StereoPannerNode(options.into()),
+            context,
+            node_options,
+            0, /* inputs */
+            1, /* outputs */
+        )?;
+        let node_id = source_node.node().node_id();
+        let pan = AudioParam::new(
+            window,
+            context,
+            node_id,
+            ParamType::Pan,
+            AutomationRate::A_rate,
+            *options.pan,
+            f32::MIN,
```

Comment:
min is `-1.`, max is `1.`

Reply:
@Manishearth Are you saying I need to change this? This is the formatting in gainnode.rs,

Reply:
Yes, from the spec the pan parameter is clamped between these values

Reply:
@Manishearth Okay, I changed it to `-1::MIN` and did the same for MAX. However, when I run `mach.bat test-wpt /webaudio/the-audio-api --log-raw log`, it says `Binary path C:\Users\Maria\servo\target\debug\servo.exe does not exist`.